fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeViewModel.java

```diff
@@ -118,44 +124,67 @@ public void handleData(String data) {
                 getApplication().getApplicationContext(), laoData.lao));
   }
 
-  protected void restoreConnections(PersistentData data) {
-    if (data == null) {
-      return;
+  /**
+   * Function to restore the state of the application.
+   *
+   * @param context activity context
+   * @return true if the wallet is correctly restored, false otherwise
+   */
+  public boolean restoreConnections(Context context) {
```

Comment:
There is no need to add the context in argument. You can use `getApplication().getApplicationContext()`